helloextend/magento-extend-integration
# feature: [PAR-4351] add post purchase lead modal

# Description

* add new template to pass parameters to `aftermarketProductProtectionModalOffer`
* add new js file to handle opening the aftermarket modal
* add new function `getLeadTokenFromUrl()` to parse the lead token from the query parameters
* update ProductProtection.php - i was running into an issue where the item added to the quote was not being correctly persisted, and from my digging it seemed like it was because the quoteId was being set on it. i believe the way it is now works because
  * the item is added to the quote and persisted
  * then when the quote is reloaded, the item is present on it and `collectTotals` will run
    * if the quote is not fetched again after it is persisted with the item, then `collectTotals` will not run - there is a boolean flag that gets set when it is reloaded, which i assume is because a new item has been loaded and the existing total does not match.

## Testing
* add a valid lead token to the cart by itself and check out -> contract is created, lead token is `consumed` ✅
* add a valid lead token to the cart, then add other items and check out -> contract is created, lead token is `consumed` ✅
* add other items, then add a lead token to the cary -> contract is created, lead token is `consumed` ✅

## Questions
* ~i noticed error `Missing required option "storeId"` occasionally occurs (in the browser console) when navigating to the store with a lead token in the query string, after clearing cookies/the cache. this is an error caused by the aftermarket modal trying to open before `'storeId'` is set in the sdk's `globalConfig` - not sure what the right resolution is there, but it's a question i had.~
  * fixed by explicitly passing the storeId into the new js function, setting to Extend.config, and explicitly passing it to the sdk call to open the modal

## Ticket

https://helloextend.atlassian.net/browse/PAR-4351

## Type of change

Please delete options that are not relevant.

- [ ] Refactor (new code changes but everything functions the same)

## Checklist

- [x] I have deployed and validated my code on a sandbox
